cpp/tests/large_strings/large_strings_tests.cpp

```diff
@@ -0,0 +1,116 @@
+
+  ~LargeStringsData() {}
+
+  void add_table(std::string_view name, std::unique_ptr<cudf::table>&& data)
+  {
+    _data[std::string(name)] = std::move(data);
+  }
+
+  cudf::table_view get_table(std::string_view name) const
+  {
+    std::string key{name};
+    return _data.find(key) != _data.end() ? _data.at(key)->view() : cudf::table_view{};
+  }
+
+  void add_column(std::string_view name, std::unique_ptr<cudf::column>&& data)
+  {
+    std::vector<std::unique_ptr<cudf::column>> cols;
+    cols.emplace_back(std::move(data));
+    _data[std::string(name)] = std::make_unique<cudf::table>(std::move(cols));
+  }
+
+  cudf::column_view get_column(std::string_view name) const
+  {
+    std::string key{name};
+    return _data.find(key) != _data.end() ? _data.at(key)->view().column(0) : cudf::column_view{};
+  }
+
+  bool has_key(std::string_view name) { return _data.find(std::string(name)) != _data.end(); }
+
+ protected:
+  std::map<std::string, DataPointer> _data;
+};
+
+cudf::column_view StringsLargeTest::wide_column()
+{
+  std::string name{"wide1"};
+  if (!g_ls_data->has_key(name)) {
+    auto input =
+      cudf::test::strings_column_wrapper({"the quick brown fox jumps over the lazy dog",
+                                          "the fat cat lays next to the other accénted cat",
+                                          "a slow moving turtlé cannot catch the bird",
+                                          "which can be composéd together to form a more complete",
+                                          "The result does not include the value in the sum in"});
+    auto counts = cudf::test::fixed_width_column_wrapper<int>({8, 8, 8, 8, 8});
+    auto result = cudf::strings::repeat_strings(cudf::strings_column_view(input), counts);
+    g_ls_data->add_column(name, std::move(result));
+  }
+  return g_ls_data->get_column(name);
+}
+
+cudf::column_view StringsLargeTest::long_column()
+{
+  std::string name("long1");
+  if (!g_ls_data->has_key(name)) {
+    auto itr = thrust::constant_iterator<std::string_view>(
+      "abcdefghijklmnopqrstuvwxyABCDEFGHIJKLMNOPQRSTUVWXY");                // 50 bytes
+    auto input = cudf::test::strings_column_wrapper(itr, itr + 5'000'000);  // 250MB
+    g_ls_data->add_column(name, input.release());
+  }
+  return g_ls_data->get_column(name);
+}
+
+LargeStringsData* StringsLargeTest::g_ls_data = nullptr;
+}  // namespace cudf::test
+
+int main(int argc, char** argv)
+{
+  ::testing::InitGoogleTest(&argc, argv);
+  auto const cmd_opts = parse_cudf_test_opts(argc, argv);
+  // hardcoding the CUDA memory resource to keep from exceeding the pool
+  auto mr = cudf::test::make_cuda();
+  rmm::mr::set_current_device_resource(mr.get());
+  auto adaptor = make_stream_mode_adaptor(cmd_opts);
+
+  // create object to automatically be destroyed at the end of main()
+  auto lsd = cudf::test::LargeStringsData();
+  // set object pointer into static variable
+  cudf::test::StringsLargeTest::g_ls_data = &lsd;
```

Comment:
Oh no, manually assigning static variable like this is not a good practice. Can we initialize it automatically?

Reply:
The issue is that the variable (or its pointer at least) needs to be accessible globally but the lifetime scope must be within `main()`. So `lsd` must be created and destroyed within the `main()` scope but needs to be singleton for the entire process at the same time.

Reply:
How about create-on-first-access?
```
struct StringsLargeTest : public cudf::test::BaseFixture {
  public:
  static auto get_ls_data() {
    g_ls_data = new cudf::test::LargeStringsData;
    return g_ls_data;
  }
  private:
  static LargeStringsData* g_ls_data;
};
```

Reply:
This won't work because it will return a pointer to main which will not automatically destroy it.

Reply:
Is there value in making this a function static?  Its construction is guaranteed to be thread safe, and will be destroyed in reverse order of construction.
```c++
static auto get_ls_data() {
  auto the_instance = cudf::test::LargeStringsData{};
  return &the_instance;
}
```

Reply:
Nope. The scope is wrong. The static object will be destroyed outside of `main()` in this case.

Reply:
Got it, so we need to create a local object in `main` and bind it to `g_ls_data`. I'm fine with the current approach. 

In case we want to do something better then:
 * Instead of accessing `g_ls_data` directly, we should access it through at getter (`get_ls_data()`).
 * The getter needs to check and throw if `g_ls_data` is not yet initialized.

Reply:
> If the object (not the pointer) is a static variable anywhere there is a chance it could be destroyed outside of main().

Sorry, I don't fully understand the concern.  The function static object is guaranteed to be alive until `main()` exits.  Does that not suit?  

Do we have a dependency somewhere in the global static destruction sequence, or something?

Reply:
> The getter needs to check and throw if g_ls_data is not yet initialized.

@ttnghia: With a function static, the object is guaranteed to be initialized once, on the first call.

Reply:
Initializing part is not too challenging. Global static destruction is not good since the object holds device memory.
Here is a godbolt which I hope will explain some of this: https://godbolt.org/z/rTa9ceEKf

Reply:
> Global static destruction is not good since the object holds device memory.

Hmm. Thank you, I'll try bear this in mind.